docs/proposals/loadbalancing.md

```diff
@@ -0,0 +1,311 @@
+* A user wants to isolate their host from decryption of other hosts saturating cpu on a single bare metal loadbalancer (this isn't really a problem for cloud) by explicitly asking for a new one.)
+
+### How does the ingress path choose a loadbalancer to join?
+
+For 1.1, there will only be one loadbalancer controller that claims all ingress paths. Whether it creates a new loadbalancer or reuses an existing one is upto the controller. For example, the GCE controller will create 1 loadbalancer per ingress path, if it has `type: network`, this will be an l4, if it has `type: application` it will be an l7 with rules to allow proxying to different backends based on the given pathmap.
+
+However we need to provide a way for users to specify which class of loadbalancers to join, and which particular loadbalancer (in the case of multiple loadbalancer per class). Note that we can't just say each class of loadbalancers has a single loadbalancer in it, and so picking a class boils down to picking a specific loadbalancer, because of cloud provider limitations. Nor can we say picking a specific loadbalancer picks the class, because a loadbalancer might have limited capacity (i.e don't force a user to pick 172.12, which might be maxed out, when they just want class=gce).
+
+The proposed long term solution for this problem is to reference a claim on an ip through the Ingress Path.
+
+### How does TLS work?
+
+Each IngressPath specifies a single secret and a tls mode. Its upto the loadbalancer controller to handle the tls mode. For example, the gce controller will only support TLS termination for 1.1 because each IngressPath gets a new loadbalancer, and GCE doesn't allow multiple certs per IP. With the current design a user can create the wrong type of secret for the wrong mode. They wouldn't know till they try creating the Ingress path. Validating a simple key:value secret is also hard.
+
+Some alternatives to the proposed tls mode handling:
+* Remove tlsMode and create a new type of secret(s).
+* Don't use a secret, embed the TLSConfig in the IngressPath.
+
+### What triggers the creation of a new loadbalancer controller, should this be exposed to users?
+
+In the current proposal, the kube-controller-manager is started up with a --loadbalancer-controllers flag that dictates which controllers to startup. This is obviously sub-optimal. A better solution would be --loadbalancer-controllers=gce-lbc-pod,haproxy-lbc-pod..., then this just boils down to creating some rcs. Taking this a step further, we could have a resource to allow dynamically adding more loadbalancer controllers to the cluster.
+
+### How does public DNS work?
+
+Most of this is TDB. With the current model, if someone specifies a hostname the loadbalancer controllers assume they know what they're doing. If there are 2 IngressPaths with the same hostname and url endpoints, one of them will win. This is just like the overlapping labels rc problem.
```

Comment:
This is ultimately a policy and security decision that admins make (an admin may want to split a workload between two namespaces at a balancer, and thus allow hosts to conflict).  A different controller may chose an ordering to impose where first comes wins.  There are too many aspects to have one solution.  Replaceable controllers is more appropriate.